tests/integration/test_external_url.py

```diff
@@ -24,18 +24,18 @@
 
 @pytest.mark.abort_on_fail
 async def test_setup_env(ops_test: OpsTest):
-    await ops_test.model.set_config(
+    await ops_test.model.set_config(  # type: ignore[union-attr]
         {"update-status-hook-interval": "60m", "logging-config": "<root>=WARNING; unit=DEBUG"}
     )
 
 
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test: OpsTest, charm_under_test):
     # deploy charm from local source folder
-    await ops_test.model.deploy(
+    await ops_test.model.deploy(  # type: ignore[union-attr]
         charm_under_test, resources=resources, application_name=app_name, trust=True
     )
-    await ops_test.model.wait_for_idle(status="active", timeout=deploy_timeout)
+    await ops_test.model.wait_for_idle(status="active", timeout=deploy_timeout)  # type: ignore[union-attr]  # noqa: E501
```

Comment:
I thought the type ignores shouldn't be needed:
https://github.com/canonical/alertmanager-k8s-operator/blob/fb9ad0012e125eac56049efe002bcf35839db1e9/pyproject.toml#L55

Feel free to update pyproject so you won't need all the ignores.

Reply:
There are 2 ways of ignoring `union-attr`:
1. In `pyproject.toml` - using `disable_error_code`. The problem is that it needs to be set globally. 
2. In `tox.ini` - using `--disable-error-code`. This allows disabling `union-attr` for specific envs only (i.e. static-integration).

I'd rather go with option 2. What do you think?

Reply:
This worked for me:
```diff
diff --git a/pyproject.toml b/pyproject.toml
index a7212de..7c9b1fc 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -52,7 +52,7 @@ allow_redefinition = true
 
 # Ignore libraries that do not have type hint nor stubs
 [[tool.mypy.overrides]]
-module = ["ops.*", "lightkube.*", "git.*", "pytest_operator.*", "validators.*"]
+module = ["ops.*", "lightkube.*", "git.*", "validators.*"]
 ignore_missing_imports = true
 
 [[tool.mypy.overrides]]
@@ -60,6 +60,11 @@ module = ["charms.grafana_k8s.*", "charms.traefik_k8s.*", "charms.observability_
 follow_imports = "silent"
 warn_unused_ignores = false
 
+[[tool.mypy.overrides]]
+module = ["pytest_operator.*"]
+follow_imports = "skip"
+ignore_missing_imports = true
+
 [tool.pytest.ini_options]
 minversion = "6.0"
 log_cli_level = "INFO"
```

Reply:
Yes, but this will ignore more than just `union-attr`. Are we ok with that?

Reply:
- I think that's what we were doing for a while anyway. [My recent contribution](https://github.com/charmed-kubernetes/pytest-operator/pull/85) didn't help much in that regard, due to the pervasive `Optional`s.
- Some (e.g. @rbarry82) argue we shouldn't even run static checks on tests.

Reply:
OK, updated.